### DISCUSSION TITLE
Is there a limit on usage by public repos?

### DISCUSSION THREAD

**Discussion by ovesh** [community/community#70492] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm a little confused by [the documentation](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions). It says:
> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories, and for self-hosted runners.

but also lists "GitHub Free" as a plan, limited to 2,000 minutes/month. Is that limit enforced on public repos?

**Comment by DNin01** [comment]
Hmm... it says here on [the Pricing page](https://github.com/pricing#compare-features) that it is free for public repositories.
![Screenshot of Compare features on Pricing page, highlighting GitHub Actions which provides 2000 minutes per month for the free plan, and minutes are free for public repositories](https://github.com/community/community/assets/106490990/8da46817-83e7-4b34-ab8e-16c628b45163)

And in the documentation page you provided a link to, the following sentences clarify that further:

Private repos in particular do indeed have a limit on how long you can use Actions for free.
> For private repositories, each GitHub account receives a certain amount of free minutes and storage for use with GitHub-hosted runners, depending on the account's plan. Any usage beyond the included amounts is controlled by spending limits.

And just above that, it explains how it is free for public repos.
> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories

Followed by more information related to private repos again.
> If you are a monthly-billed customer, your account will have a default spending limit of 0 US dollars (USD), which prevents additional usage of minutes or storage **for private repositories** beyond the amounts included with your account.

This information seems to imply that yes, the standard runners for GitHub Actions are completely free for public repositories, no matter how long they work for. I don't manage any actions so I can't confirm for sure, but it looks like this is correct.

**Comment by ovesh** [comment, marked as answer]
I got official confirmation from Github support that 
1. Public repos do not have any minute limitation
2. This includes Windows/MacOS runners, as long as you use the standard runners where "standard" applies to CPU/ram/disk, i.e not ["larger runners"](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#title-h1).

**Reply by madhuryadutta** [reply]
Still there is  a **execution limit** per **Job Run** .
I got the following **error** after running **github hosted runner for 6hour (360 minutes)** .
![Screenshot from 2024-01-08 00-20-24](https://github.com/community/community/assets/97863234/026eb1ff-ba98-4025-82fe-72b2fd23a334)

After searching for the reason I find the following Information
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

![Screenshot from 2024-01-08 00-19-37](https://github.com/community/community/assets/97863234/fb64cf0a-9ce0-428a-bee6-beee15d227bc)

 The following answer says that **self hosted runner** can be run upto **72 hour** and **github hosted runner** for **6 hour**.
https://github.com/orgs/community/discussions/27177

**Reply by op30mmd** [reply]
> Still there is  a **execution limit** per **Job Run** .
> I got the following **error** after running **github hosted runner for 6hour (360 minutes)** .
> ![Screenshot from 2024-01-08 00-20-24](https://github.com/community/community/assets/97863234/026eb1ff-ba98-4025-82fe-72b2fd23a334)
> 
> After searching for the reason I find the following Information
> https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes
> 
> ![Screenshot from 2024-01-08 00-19-37](https://github.com/community/community/assets/97863234/fb64cf0a-9ce0-428a-bee6-beee15d227bc)
> 
>  The following answer says that **self hosted runner** can be run upto **72 hour** and **github hosted runner** for **6 hour**.
> https://github.com/orgs/community/discussions/27177
> 
> 
> 
> 
> 
> 
> 
> 

@madhuryadutta simply use schedule

**Reply by fallenreaper** [reply]
This is in part because if you can create something that runs forever, then you essentially can use this to create a server that runs indef.  Having limits on runner time prevent you from essentially launching and running a server that can run indef, processing data etc.  These limits help prevent those sort of things because while they arent meant for it, it is something that can easily be rigged to do.

**Reply by rafark** [reply]
> Still there is a **execution limit** per **Job Run** . I got the following **error** after running **github hosted runner for 6hour (360 minutes)** . ![Screenshot from 2024-01-08 00-20-24](https://private-user-images.githubusercontent.com/97863234/294779408-026eb1ff-ba98-4025-82fe-72b2fd23a334.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODYwODk5NjUsIm5iZiI6MTc4NjA4OTY2NSwicGF0aCI6Ii85Nzg2MzIzNC8yOTQ3Nzk0MDgtMDI2ZWIxZmYtYmE5OC00MDI1LTgyZmUtNzJiMmZkMjNhMzM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODA3VDA4MDEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5NDYzODI5OGZiYTRhZDFkZGRjOTNhZTQzZTcyNDE3NDFmNjg2NDlhMmMzNWEzYmE0YTUwMjAyNDZiYzgwODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.Qws0xLUUXdU_8l3KNMN8st3ufiWL32Q6ZHNxS0JZBTM)
> 
> After searching for the reason I find the following Information https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes
> 
> ![Screenshot from 2024-01-08 00-19-37](https://private-user-images.githubusercontent.com/97863234/294779401-fb64cf0a-9ce0-428a-bee6-beee15d227bc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODYwODk5NjUsIm5iZiI6MTc4NjA4OTY2NSwicGF0aCI6Ii85Nzg2MzIzNC8yOTQ3Nzk0MDEtZmI2NGNmMGEtOWNlMC00MjhhLWJlZTYtYmVlZTE1ZDIyN2JjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODA3VDA4MDEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3MTVhYWYzNGQ4MGJmNjk5MzAyZDcyYzEyYWI4Y2QyMGY0Y2Q3ZDlhMGM1ZDMzNzBkM2I0MzhkNzI4N2Y3M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.tFMUWW0pnTqe6Sm2xVozNnTp4rG5wLCLmqyqyk0F40Q)
> 
> The following answer says that **self hosted runner** can be run upto **72 hour** and **github hosted runner** for **6 hour**. https://github.com/orgs/community/discussions/27177

My brother in Christ 6 hours? Who needs to run a job for that long? That seems like a hard limit to prevent abuse. As a matter of fact I’m surprised the cap is 6 hours. GitHub actions are not designed to run that long

**Reply by MathijsNabbe** [reply]
> My brother in Christ 6 hours? Who needs to run a job for that long? That seems like a hard limit to prevent abuse. As a matter of fact I’m surprised the cap is 6 hours. GitHub actions are not designed to run that long

Exactly my point, i once had a job running for 35 minutes and i thought that was waaaayyy too long. 6 hours? Damn.

**Reply by Malix-Labs** [reply]
This happens when you run extensive compilation or test suites, including a custom Linux for example.

The nixpkgs / nixos repository (which is the biggest in the world and the one with the most extensive CI/CD) can run for days

**Comment by GCC1111** [comment]
@DNin01
